frontend/components/ColorPicker/ColorPicker.jsx

```diff
@@ -0,0 +1,201 @@
+'use strict'
+
+import {h, Component} from 'preact'
+import proptypes from 'proptypes'
+
+import Style from 'lib/Style'
+import styles from './ColorPicker.css'
+
+import Color from 'lib/color'
+
+/**
+ * Dialog for picking date and time for an input field.
+ */
+export default class ColorPicker extends Component {
+  static propTypes = {
+    /**
+     * Classes to append to the button element.
+     */
+    className: proptypes.string,
+
+    /**
+     * The Color string representing the currently selected value.
+     */
+    color: proptypes.string,
+
+    /**
+     * The format of the color string
+     */
+    format: proptypes.string,
+
+    /**
+     * A callback to be fired when a new date is selected.
+     */
+    onChange: proptypes.func
+  }
+
+  constructor(props) {
+    super(props)
+
+    this.hsv = false
+
+    this.hueCoordinate = false
+    this.hueElement = false
+    this.hueElementHeight = false
+    this.hueIndicator = false
+
+    this.paletteBackground = false
+    this.paletteCoordinate = false
+    this.paletteElement = false
+    this.paletteIndicator = false
+  }
+
+  componentDidMount() {
+    this.setInitial()
```

Comment:
`componentDidMount` will fire when the component is initialised, after the first `render`. But you're also calling `this.setInitial` on `render`, so it will be called twice when the component is mounted. Is this intentional? Could you not remove `componentDidMount` entirely?